tests/Integration/Behaviour/Features/Context/Domain/AbstractDomainFeatureContext.php

```diff
@@ -293,6 +294,14 @@ protected function localizeByCell(string $localizedValue): array
         return $localizedValues;
     }
 
+    /**
+     * @return int
+     */
+    protected function getContextShopId(): int
```

Comment:
We shouldn't integrate anything fetched from the Context in our behat tests **ever**

The reason is that we are currently "struggling" to change the command format to force them and include some ShopConstraint exactly because we don't want to rely on the Context anymore So if we get rid of it somewhere it's not to introduce it again somewhere else

The command are now built to have a full knowledge of their own context, so their tests should as well. So this should should be removed.

When you're not sure which shopId to use in your behat tests that are oriented as single shop you can usually use the default shop.

You can use `AbstractProductFeatureContext::getDefaultShopId()` in those case (maybe one day we'll have to move it to a lower layer like `AbstractDomainFeatureContext` but for now it's mainly useful for product commands which are the only ones handling shop constraints (for now)

Reply:
Ah I just noticed that you need this in `CategoryFeatureContext` which doesnextend the product class, so I guess now is a good time to move `getDefaultShopId` to `AbstractDomainFeatureContext` ^^